### PR TITLE
Sort matrix java

### PR DESCRIPTION
**Pull request by sohini3062**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?
Yes
(Write your answer here.)

### Description
Simple sorting of matrix in Java
(Write your answer here.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
Add matrix sorting in Java #2273
(Write your answer here.)

### CODE REVIEW

File: Java/sort/SORT_MATRIX.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
+package com.company;
```
Comment (HarshCasper) [inline Java/sort/SORT_MATRIX.java]: Remove Package

File: Java/sort/SORT_MATRIX.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
+package com.company;
+
+
+import java.util.*;
+
+class Main {
+    public static void main(String[] args)
+    {
+        /*Sorting a matrix may look tough but you have to view a matrix as a collection of arrays. If you see it as an array, you can sort it by using any custom sorting algorithm, I will be using Arrays.sort(ARRAY_NAME) here which uses QuickSort. If we copy the matrix elements to a temporary array of size number of rows*number of columns of the matrix at hand and sort it and copy it back, we can get the sorted matrix. The approach may seem to be simple but it is a good practice for beginners to know they can solve problems by brute force at first before optimizing the code*/
```
Comment (HarshCasper) [inline Java/sort/SORT_MATRIX.java]: Make it compact to read and review

File: Java/sort/SORT_MATRIX.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        //taking the matrix as input taking O(n*m) time complexity where n=rows and m=cols
+        for(int i=0;i<rows;i++)
+        {
+            for(int j=0;j<cols;j++)
+                matrix[i][j]=abc.nextInt();
+        }
+        //creating a temporary 1D array
+        int temporary[]=new int[rows*cols];
+        //display the matrix and copy it to the temp array
+        int k=0;//iterating and storing elements in the temp array
```
Comment (HarshCasper) [inline Java/sort/SORT_MATRIX.java]: Use comments above the line of code not beside it

File: Java/sort/SORT_MATRIX.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+23  99  1234
+89  1  0
+23  -99  -345
+
+-345  -99  -45
+0  0  1
+23  23  34
+89  99  1234   */
+    }
+}
```
Comment (HarshCasper) [inline Java/sort/SORT_MATRIX.java]: Indent it properly and make it look presentable

File: Java/sort/SORT_MATRIX.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+                matrix[i][j]=temporary[k++];
+
+        }
+        //display final state of original matrix
+        for(int i=0;i<rows;i++)
+        {
+            for(int j=0;j<cols;j++) {
+                System.out.print(matrix[i][j]+"  ");
+            }
+            System.out.println();
```
Comment (HarshCasper) [inline Java/sort/SORT_MATRIX.java]: Format the Code. Use any online Java Code Formatter for the purpose

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: SORT_MATRIX.java [hunk 6]
```diff
@@ -0,0 +1,85 @@
+import java.util.*;
```
Comment (HarshCasper) [inline SORT_MATRIX.java:1]: Why has this Code been pushed on the root?

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: Java/sort/SORT_MATRIX.java (hunk 1 above)
Comment (HarshCasper) [reply Java/sort/SORT_MATRIX.java]: Not yet resolved.

File: Java/sort/SORT_MATRIX.java (hunk 2 above)
Comment (HarshCasper) [reply Java/sort/SORT_MATRIX.java]: Not yet resolved.

File: Java/sort/SORT_MATRIX.java (hunk 3 above)
Comment (HarshCasper) [reply Java/sort/SORT_MATRIX.java]: Not yet resolved.

File: Java/sort/SORT_MATRIX.java (hunk 4 above)
Comment (HarshCasper) [reply Java/sort/SORT_MATRIX.java]: Not yet resolved.

File: Java/sort/SORT_MATRIX.java (hunk 5 above)
Comment (HarshCasper) [reply Java/sort/SORT_MATRIX.java]: Not yet resolved.

File: Java/sort/SORT_MATRIX.java (hunk 1 above)
Comment (sohini3062) [reply Java/sort/SORT_MATRIX.java]: I have added the new code. Can you check?

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]
@sohini3062  You are sorting the matrix only row-wise. Try to consider both cases.
Better to include sorting technique for column-wise also.

**Comment by atarax665** [comment]
No Updates.

**Pull request closed without merging**